hack/tidy-check.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+# Copyright 2019 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set +e
+THIS_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+PROJECT_DIR="$THIS_DIR/.."
+pushd "$THIS_DIR" >/dev/null || exit
+MOD_FILE="$THIS_DIR/../go.mod"
+SUM_FILE="$THIS_DIR/../go.sum"
+TMP_DIR="$THIS_DIR/.tmp.tidy-check"
+TMP_MOD_FILE="$TMP_DIR/go.mod"
+TMP_SUM_FILE="$TMP_DIR/go.sum"
+TARGET_GO_VER="go1.12.*"
+
+general_help() {
+  echo "Please run the following command to generate a new go.mod & go.sum:"
+  if [[ "$(go version | awk '{print $3}')" == $TARGET_GO_VER ]]; then
+    echo "  \$ make tidy"
+  else
+    echo "  \$ make docker-tidy"
+  fi
+}
+
+precheck() {
+  if [ ! -r "$MOD_FILE" ]; then
+    echo "no go.mod found"
+    general_help
+    exit 1
+  fi
+  if [ ! -r "$SUM_FILE" ]; then
+    echo "no go.sum found"
+    general_help
+    exit 1
+  fi
+  mkdir -p "$TMP_DIR"
+}
+
+tidy() {
+  cp "$MOD_FILE" "$TMP_MOD_FILE"
+  mv "$SUM_FILE" "$TMP_SUM_FILE"
+  if [[ "$(go version | awk '{print $3}')" == $TARGET_GO_VER ]]; then
```

Comment:
The Makefile lets you set the `GO` environment variable so maybe we should accommodate that case. In theory I can have multiple Go versions installed and pick the one I want to use for Antrea by setting the `GO` env variable. The one I pick may not be the default version which is used when I run `go`.

Reply:
Thanks Antonin, I have enhanced accordingly, PTAL.